src/measure_theory/measure/with_density_vector_measure.lean

```diff
@@ -0,0 +1,140 @@
+/-
+Copyright (c) 2021 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import measure_theory.measure.vector_measure
+import measure_theory.function.ae_eq_of_integral
+
+/-!
+
+# Vector measure defined by an integral
+
+Given a measure `μ` and an integrable function `f : α → E`, we can define a vector measure `v` such
+that for all measurable set `s`, `v i = ∫ x in s, f x ∂μ`. This definition is useful for
+the Radon-Nikodym theorem for signed measures.
+
+## Main definitions
+
+* `measure_theory.measure.with_densityᵥ`: the vector measure formed by integrating a function `f`
+  with respect to a measure `μ` on some set if `f` is integrable, and `0` otherwise.
+
+-/
+
+noncomputable theory
+open_locale classical measure_theory nnreal ennreal
+
+variables {α β : Type*} {m : measurable_space α}
+
+namespace measure_theory
+
+open topological_space
+
+variables {μ ν : measure α}
+variables {E : Type*} [normed_group E] [measurable_space E] [second_countable_topology E]
+  [normed_space ℝ E] [complete_space E] [borel_space E]
+
+/-- Given a measure `μ` and an integrable function `f`, `μ.with_densityᵥ f` is
+the vector measure which maps the set `s` to `∫ₛ f ∂μ`. -/
+def measure.with_densityᵥ {m : measurable_space α} (μ : measure α) (f : α → E) :
+  vector_measure α E :=
+if hf : integrable f μ then
+{ measure_of' := λ s, if measurable_set s then ∫ x in s, f x ∂μ else 0,
+  empty' := by simp,
+  not_measurable' := λ s hs, if_neg hs,
+  m_Union' := λ s hs₁ hs₂,
+  begin
+    convert has_sum_integral_Union hs₁ hs₂ hf,
+    { ext n, rw if_pos (hs₁ n) },
+    { rw if_pos (measurable_set.Union hs₁) }
+  end }
+else 0
+
+open measure
+
+include m
+
+variables {f g : α → E}
+
+lemma with_densityᵥ_apply (hf : integrable f μ) {s : set α} (hs : measurable_set s) :
+  μ.with_densityᵥ f s = ∫ x in s, f x ∂μ :=
+by { rw [with_densityᵥ, dif_pos hf], exact dif_pos hs }
+
+@[simp]
+lemma with_densityᵥ_zero : μ.with_densityᵥ (0 : α → E) = 0 :=
```

Comment:
```suggestion
@[simp] lemma with_densityᵥ_zero : μ.with_densityᵥ (0 : α → E) = 0 :=
```
Same for the other lemmas. Let's make the file shorter when we can. It makes it easier to read.